Swaggeriffic

cmason opened the pull request:
Document api with swagger.

I'm on the fence about adding a DSL to models and controllers to generate the swagger.json file. So I kept everything static for now. `swagger-v1.json` can be easily edited with swagger-editor.

![screen shot 2017-09-09 at 6 35 32 pm](https://user-images.githubusercontent.com/47318/30245050-63aec9c6-9593-11e7-967d-88f5f6bf0cc5.png)

cmason committed:
API documentation with swagger

cmason committed:
Update swagger json file to where it will be once in master

ironcoconut commented:
Looks good. I'll check out this branch in the afternoon and play with it.

I'd like to add v2 support in a future PR. Can you link me to the relevant swagger docs?

jwo commented:
@cmason do you think we should link to the raw swagger json on master? or should we store the swagger in in like:

* `/api/v2/index.html` (docs) and `/api/v2/swagger.json`
* `/api/v2/index.html` (docs) and `/api/v2/swagger.json`

I'm good with this, seems accurate. Would like to discuss above and then merge and then update readme to direct people to the live docs

cmason commented:
I like your idea better. It allows for the documentation to match the current branch and not just what's in master

jwo commented:
@cmason ✔️ 

If you can make that change, super cool, let's murg. If not, I can pull this code into a branch and get it live.

cmason commented:
I'll get that change in tonight and throw in the v2 version as well.


On Mon, Sep 11, 2017 at 2:42 PM Jesse Wolgamott <notifications@github.com>
wrote:

> @cmason <https://github.com/cmason> ✔️
>
> If you can make that change, super cool, let's murg. If not, I can pull
> this code into a branch and get it live.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sketch-city/harvey-api/pull/102#issuecomment-328637268>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAC41sZ78OKl2xdervBFVZKRSZWS1O2xks5shY0ZgaJpZM4PSKOC>
> .
>
-- 
*Chris Mason*    Software Architect
*www.chaione.com <http://www.chaione.com/>* | O: *888-316-0357
<//888-316-0357;102>* | @chaione <https://twitter.com/chaione>
1800 West Loop South #2100, Houston, TX 77027

*ChaiOne:* Digital Innovation Agency

cmason committed:
Keep swagger data files relative to html

Allows branches to display their curent documentation as opposed
to only what is in master.

cmason committed:
Merge remote-tracking branch 'upstream/master' into swaggeriffic

cmason committed:
Added documentation for V2

Could stand to be improved!

cmason commented:
@jwo I moved the swagger definition files to sit next to the documentation page. This allows the documentation to match whatever branch you're on.

The `v2` api is now documented, just the bare minimum (only 15 pieces of flair). You should be able to toggle back and forth between the two with the select box at the top.

New versions of the api will have to be added to `index.html` manually. I didn't bother automating it (rule of three and all).

jwo committed:
Merge branch 'master' into swaggeriffic

jwo reviewed (approved):
Great job!
